templates/default.nginx_http.conf.template

```diff
@@ -21,6 +21,20 @@ server_tokens off;
 # Don't allow page to be rendered inside a frame or iframe
 add_header X-Frame-Options DENY;
 
+# Enable cross-site scripting filter (XSS)
+# (handled by the browser; usually enabled by default,
+# this header will re-enable the filter if disabled by user.)
+add_header X-XSS-Protection "1; mode=block";
+
+# Explicitly list domains allowed to serve content for this site
+add_header Content-Security-Policy-Report-Only "default-src 'self'; object-src 'none'; base-uri 'self'; manifest-src 'self'; media-src 'self'; worker-src 'none';";
+add_header Content-Security-Policy-Report-Only "script-src 'report-sample' 'self' https://discuss.dockstore.org https://gui.dockstore.org https://platform.twitter.com https://www.google-analytics.com https://www.googletagmanager.com;";
+add_header Content-Security-Policy-Report-Only "style-src 'report-sample' 'self' https://cdnjs.cloudflare.com https://fonts.googleapis.com https://gui.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "connect-src 'self' https://s3.amazonaws.com https://api.github.com https://view.commonwl.org;";
+add_header Content-Security-Policy-Report-Only "font-src 'self' https://fonts.gstatic.com https://gui.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "frame-src 'self' https://discuss.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "img-src 'self' data: https://avatars0.githubusercontent.com https://avatars1.githubusercontent.com https://avatars3.githubusercontent.com https://gui.dockstore.org https://i.imgur.com https://img.shields.io https://quay.io https://via.placeholder.com https://www.googletagmanager.com https://www.gravatar.com;";
```

Comment:
github might actually have a doc for where they serve content which might be a useful code comment here

Reply:
or maybe `https://*.githubusercontent.com`, since that will match githubusercontent.com and any other subdomains ([ref](https://w3c.github.io/webappsec-csp/#source-expression))

Reply:
Was not able to find any document listing places where Github stores its content, so I just changed the githubusercontent.com domains to use wildcards.

Reply:
I think we want to specify the exact subdomains? I'm not sure of the maintenance ramifications, but one should not be able to link from `raw.github...` because then we are opening up any content on github to pass the security policy.

Reply:
See above - thanks for raising the point about raw.githubusercontent.com. Have updated CSP to explicitly list subdomains of githubusercontent.com instead of using a wildcard.